staging/src/k8s.io/apiserver/pkg/server/resourceconfig/helpers.go

```diff
@@ -78,9 +78,9 @@ func MergeAPIResourceConfigs(
 	if ok {
 		if allAPIFlagValue == "false" {
 			// Disable all group versions.
-			resourceConfig.DisableVersions(registry.RegisteredGroupVersions()...)
+			resourceConfig.DisableAll()
 		} else if allAPIFlagValue == "true" {
-			resourceConfig.EnableVersions(registry.RegisteredGroupVersions()...)
+			resourceConfig.EnableAll()
```

Comment:
why this? If so, we must explicitly  all group/versions in https://github.com/kubernetes/kubernetes/blob/master/pkg/master/master.go#L461  Otherwise, api/all=true may not enable all group versions.

Reply:
> why this? If so, we must explicitly all group/versions in https://github.com/kubernetes/kubernetes/blob/master/pkg/master/master.go#L461 Otherwise, api/all=true may not enable all group versions.

You must explicitly choose on or off for the items you want to serve. This much more specific than trying to infer what may be served based on a type definition which doesn't necessarily serve all of the types it contains.